.github/workflows/publish.yaml

```diff
@@ -0,0 +1,40 @@
+name: Publish
+
+on: workflow_dispatch
+
+jobs:
+  cargo-publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.version_tag }}
+
+      - name: Fetch tags
+        run: git fetch --prune --unshallow --tags
```

Comment:
Default checkout is shallow last-version-only. This also fetches all the tags. Previous version tried to clone the whole repo every time.